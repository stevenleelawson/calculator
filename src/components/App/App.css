.App {
  text-align: center;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: chartreuse;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  margin: 3em;
}
