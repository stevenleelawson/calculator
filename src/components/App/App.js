import React, { Component } from 'react';
import './App.css';
import ButtonContainer from '../ButtonContainer';
import Display from '../Display';

class App extends Component {
  constructor() {
    super();
    this.state = {
      numbers: [],
    }
  }

  handleClick = (e) => {
    let value = e.target.getAttribute('value');
    this.setState({
      numbers: [...this.state.numbers, value]
    });
  }

  handleEvaluateClick = () => {
    let joinedNums = this.state.numbers.join('');
    let result = eval(joinedNums)

    this.setState({
      numbers: [result]
    });
  }

  handleClearClick = () => {
    this.setState({
      numbers: [],
    });
  }

  handleDelClick = () => {
    this.state.numbers.pop()
    this.setState({
      numbers: [...this.state.numbers]
    });
  }

  handleSqClick = () => {
    let joinedNums = Math.sqrt(this.state.numbers.join(''))
    this.setState({
      numbers: [joinedNums]
    });
  }

  handlePercentClick = () => {
    const joinedNums = this.state.numbers.join('');
    const percentNum = joinedNums * .01;
    this.setState({
      numbers: [percentNum]
    });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Steven's Calculator</h1>
          <Display displayedNumbers={this.state.numbers} />
        </header>
        <div>
          <div  className="container">
            <ButtonContainer
              handleClick={this.handleClick} handleEvaluateClick={this.handleEvaluateClick}
              handleClearClick={this.handleClearClick}
              handleDelClick={this.handleDelClick}
              handleSqClick={this.handleSqClick}
              handlePercentClick={this.handlePercentClick}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
