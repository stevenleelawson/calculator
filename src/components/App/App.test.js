import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
    wrapper.setState({
      numbers: ["2","+","2"]
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('evaluates operations in state', () => {

    wrapper.instance().handleEvaluateClick();

    expect(wrapper.state('numbers')).toEqual([4])
  });

  it('clears state on click of clear button', () => {

    wrapper.instance().handleClearClick();

    expect(wrapper.state('numbers')).toEqual([])
  });

  it('removes item from state when delete button clicked', () => {

    wrapper.instance().handleDelClick();

    expect(wrapper.state('numbers')).toEqual(["2","+"])
  });

  it('finds percent when % button is clicked', () => {
    wrapper.setState({
      numbers: ["4"]
    });

    wrapper.instance().handlePercentClick();

    expect(wrapper.state('numbers')).toEqual([.04])
  });

  it('finds square root when sqrt button is clicked', () => {
    wrapper.setState({
      numbers: ["4"]
    });
    wrapper.instance().handleSqClick();

    expect(wrapper.state('numbers')).toEqual([2])
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
