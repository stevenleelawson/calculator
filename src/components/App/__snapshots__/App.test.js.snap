// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Steven's Calculator
          </h1>
          <Display
                    displayedNumbers={
                              Array [
                                        "2",
                                        "+",
                                        "2",
                                      ]
                    }
          />
</header>,
        <div>
          <div
                    className="container"
          >
                    <ButtonContainer
                              handleClearClick={[Function]}
                              handleClick={[Function]}
                              handleDelClick={[Function]}
                              handleEvaluateClick={[Function]}
                              handlePercentClick={[Function]}
                              handleSqClick={[Function]}
                    />
          </div>
</div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              Steven's Calculator
</h1>,
            <Display
              displayedNumbers={
                            Array [
                                          "2",
                                          "+",
                                          "2",
                                        ]
              }
/>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Steven's Calculator",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Steven's Calculator",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "displayedNumbers": Array [
                "2",
                "+",
                "2",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <ButtonContainer
                        handleClearClick={[Function]}
                        handleClick={[Function]}
                        handleDelClick={[Function]}
                        handleEvaluateClick={[Function]}
                        handlePercentClick={[Function]}
                        handleSqClick={[Function]}
            />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ButtonContainer
              handleClearClick={[Function]}
              handleClick={[Function]}
              handleDelClick={[Function]}
              handleEvaluateClick={[Function]}
              handlePercentClick={[Function]}
              handleSqClick={[Function]}
/>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleClearClick": [Function],
              "handleClick": [Function],
              "handleDelClick": [Function],
              "handleEvaluateClick": [Function],
              "handlePercentClick": [Function],
              "handleSqClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Steven's Calculator
            </h1>
            <Display
                        displayedNumbers={
                                    Array [
                                                "2",
                                                "+",
                                                "2",
                                              ]
                        }
            />
</header>,
          <div>
            <div
                        className="container"
            >
                        <ButtonContainer
                                    handleClearClick={[Function]}
                                    handleClick={[Function]}
                                    handleDelClick={[Function]}
                                    handleEvaluateClick={[Function]}
                                    handlePercentClick={[Function]}
                                    handleSqClick={[Function]}
                        />
            </div>
</div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                Steven's Calculator
</h1>,
              <Display
                displayedNumbers={
                                Array [
                                                "2",
                                                "+",
                                                "2",
                                              ]
                }
/>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Steven's Calculator",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "Steven's Calculator",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "displayedNumbers": Array [
                  "2",
                  "+",
                  "2",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <ButtonContainer
                            handleClearClick={[Function]}
                            handleClick={[Function]}
                            handleDelClick={[Function]}
                            handleEvaluateClick={[Function]}
                            handlePercentClick={[Function]}
                            handleSqClick={[Function]}
              />
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ButtonContainer
                handleClearClick={[Function]}
                handleClick={[Function]}
                handleDelClick={[Function]}
                handleEvaluateClick={[Function]}
                handlePercentClick={[Function]}
                handleSqClick={[Function]}
/>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleClearClick": [Function],
                "handleClick": [Function],
                "handleDelClick": [Function],
                "handleEvaluateClick": [Function],
                "handlePercentClick": [Function],
                "handleSqClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
